### ISSUE TITLE
virtualenv generator with `set -u` leads to unbound variable failure

### ISSUE THREAD

**Issue by puetzk** [conan-io/conan#3911]
Conan 1.9.0, ubuntu 16.04

If a package sets env_info to a list (so that [tools.environment_append](https://docs.conan.io/en/latest/reference/tools.html?highlight=environment_append#tools-environment-append) will prepend it with a delimter, the virtualenv generator produces an assignment like:

`PYTHONPATH=".../release/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/python":$PYTHONPATH`

If there is no current value in that variable, and set -u is in effect, this will cause the shell to error out with `bash: PYTHONPATH: unbound variable`

To fix this, the virtualenv generators should explicitly use a conditional expansion like ${PYTHONPATH:-}, that can substitute to nothing if the variable is unset (even with set -u, since this was explicitly requested). But since I think it would also be nice to not get a junk trailing delimiter when there's nothing to append, I suggest:

`val1:val2${NAME+:$NAME}`

Despite being a little fancy, `${VAR:-}` and `$:{VAR+}` are portable posix sh substitutions, not bashisms (see [section 26.2 Parameter Expansion](http://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02))

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by lasote** [comment]
Probably it is better to not set the variable if we detect it is empty.

**Comment by puetzk** [comment]
No, I'm talking about when the ConanFile.env has stuff to prepend to a list-variable, but there's no current entries. You do need to set the variable, to add the additional entries that the profile/recipes provided. But right now sourcing the virtualenv scripts errors out (if you have sh set to make unset variable substitutions as an error) when there's nothing to prepend the new values in front of. Or, if you don't have that set as an error, it adds a trailing delimiter on the list of new values anyway, which could plausibly be read as the list having an empty string as the last element. 

Using `${NAME+:$NAME}` fixes both problems; it explicitly handles the case where NAME is previously unset, and nothing is appended (not even the delimiter) if name is previously unset or empty.
